### PR TITLE
REVSDL-2457 On default service chosen notification GPB implementation

### PR DESCRIPTION
**Pull request by thinhts**
Please review
@DKlimenko 
@VladSemenyuk 
@KKarlash 
@bonewell 
@giangtqh

### CODE REVIEW

File: src/components/vr_cooperation/include/vr_cooperation/commands/on_default_service_chosen_notification.h [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+#include "vr_cooperation/interface/hmi.pb.h"
+
+namespace vr_cooperation {
+
+namespace commands {
+
+/**
+ * @brief OnDefaultServiceChosenNotification command class
+ */
+class OnDefaultServiceChosenNotification : public RequestFromHMI {
```
Comment (giangtqh) [inline src/components/vr_cooperation/include/vr_cooperation/commands/on_default_service_chosen_notification.h]: notification not inherit from Request, just inherit from Command

File: src/components/vr_cooperation/include/vr_cooperation/commands/on_default_service_chosen_notification.h [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+ public:
+  /**
+   * @brief OnDefaultServiceChosenNotification class constructor
+   * @param parent pointer to VRModule
+   * @param message Message from HMI
+   **/
```
Comment (giangtqh) [inline src/components/vr_cooperation/include/vr_cooperation/commands/on_default_service_chosen_notification.h]: */

**Comment by giangtqh** [comment]
reviewed

File: src/components/vr_cooperation/src/on_default_service_chosen_notification.cc [hunk 3]
```diff
@@ -43,34 +44,31 @@ namespace commands {
...
   LOG4CXX_AUTO_TRACE(logger_);
 
   int32_t app_id = -1;
-  vr_hmi_api::ServiceMessage message = GetMessage();
   vr_hmi_api::OnDefaultServiceChosenNotification params;
-  if (message.has_params()) {
-    bool ret = params.ParseFromString(message.params());
+  if (message_.has_params()) {
+    bool ret = params.ParseFromString(message_.params());
     if (ret) {
```
Comment (giangtqh) [inline src/components/vr_cooperation/src/on_default_service_chosen_notification.cc]: `if (message_.has_params() && params.ParseFromString(message.params())`

File: src/components/vr_cooperation/src/hmi_command_factory.cc [hunk 4]
```diff
@@ -50,6 +51,8 @@ commands::Command* HMICommandFactory::CreateCommand(
 //    case vr_hmi_api::RPCName::ON_DEACTIVATED: {
 //      return new commands::OnServiceDeactivatedNotification(parent, message);
 //    }
+    case vr_hmi_api::ON_DEFAULT_CHOSEN:
+      return new commands::OnDefaultServiceChosenNotification(parent, message);
```
Comment (bonewell) [inline src/components/vr_cooperation/src/hmi_command_factory.cc]: Who deletes this object? I mean you possible have memory leak.

File: src/components/vr_cooperation/include/vr_cooperation/commands/on_default_service_chosen_notification.h [hunk 5]
```diff
@@ -34,24 +34,25 @@
...
  public:
   /**
    * @brief OnDefaultServiceChosenNotification class constructor
    * @param parent pointer to VRModule
    * @param message Message from HMI
-   **/
+   */
   OnDefaultServiceChosenNotification(
       VRModule* parent,
       const vr_hmi_api::ServiceMessage& message);
```
Comment (giangtqh) [inline src/components/vr_cooperation/include/vr_cooperation/commands/on_default_service_chosen_notification.h]: these code can put in 2 lines

**Comment by bonewell** [comment]
Reviewed.

**Comment by giangtqh** [comment]
reviewed

File: src/components/vr_cooperation/include/vr_cooperation/commands/on_default_service_chosen_notification.h (hunk 5 above)
Comment (bonewell) [inline src/components/vr_cooperation/include/vr_cooperation/commands/on_default_service_chosen_notification.h]: If you use Eclipse you can use format rules from Google for Eclipse. https://github.com/google/styleguide/blob/gh-pages/eclipse-cpp-google-style.xml

**Comment by thinhts** [comment]
@giangtqh 
@bonewell 
I fixed your comments. Please review

**Comment by giangtqh** [comment]
reviewed

**Comment by bonewell** [comment]
Reviewed.

**Comment by DKlimenko** [comment]
reviewed

File: src/components/vr_cooperation/include/vr_cooperation/commands/on_default_service_chosen_notification.h [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+
+  /**
+   * @brief run command
+   */
+  virtual void Run();
+
+  /**
+   * @brief on timeout reaction
+   */
+  virtual void OnTimeout();
```
Comment (KKarlash) [inline src/components/vr_cooperation/include/vr_cooperation/commands/on_default_service_chosen_notification.h]: OnTimeout is not necessary for notification, that's why please make another base class special for notifications

**Comment by VladSemenyuk** [comment]
Reviewed

**Comment by thinhts** [comment]
@KKarlash I change to inherit another base class for notification. This will be implement by Giang

**Comment by KKarlash** [comment]
Reviewed

**Pull request merged**